src/components/obs/ObsHistory.jsx

```diff
@@ -117,6 +119,19 @@ class ObsHistory extends React.PureComponent {
       : formatDate(this.getDateFromObs(obs));
   }
 
+  getEncounterFromObsByEncounterAndGroup(obsByEncounterAndGroup) {
```

Comment:
I haven't been 100% happy with how the ObsHistory component code reads (I wrote it a few months ago and you can see some of my concerns in the comments)... in this method and the two below I just extracted out some of the functionality in a small attempt to hopefully make it a little more clear. :)

Reply:
Hmmm... yeah the impulse is good, but I feel like the two functions that follow this one (`gitFirst...` and `flattenObs...`) are pretty superfluous.

Reply:
I wanted the "getFirst..." because it doesn't seem immediately obvious that "obsByEncounterAndGroup[0][0]" means "the first obs in the nested set"... flatten is superflous, I agree... when I wrote it I didn't realize there was a one-line Javascript method to do exactly what I wanted... :)